test/robot/monitor/monitor.go

```diff
@@ -121,27 +121,52 @@ func Run(ctx context.Context, managers Managers, owner DataOwner, update func(ct
 }
 
 func monitor(ctx context.Context, managers *Managers, owner DataOwner) error {
+	initial := &search.Query{}
 	// TODO: care about monitors erroring
-	all := &search.Query{Monitor: true}
+	monitor := &search.Query{Monitor: true}
 	if managers.Job != nil {
-		crash.Go(func() { managers.Job.SearchDevices(ctx, all, owner.updateDevice) })
-		crash.Go(func() { managers.Job.SearchWorkers(ctx, all, owner.updateWorker) })
+		if err := managers.Job.SearchDevices(ctx, initial, owner.updateDevice); err != nil {
```

Comment:
I'm a little upset that I can't find a way to refactor this into something more compact without resorting to reflection. :(

Reply:
Yeah, me too, I had written like three different solutions but this was by far the most succinct, all the rest kind of exploded into lots of code changes.